core/modules/date/date.elements.inc

```diff
@@ -852,10 +852,20 @@ function date_default_value($field, $instance, $langcode) {
   $item = array();
   $db_format = date_type_format($field['type']);
   $date = date_default_value_part($item, $field, $instance, $langcode, 'value');
-  $item[0]['value'] = is_object($date) ? date_format($date, $db_format) : '';
+  if (is_object($date) && empty($date->errors)) {
+    $item[0]['value'] = date_format($date, $db_format);
+  }
+  else {
+    $item[0]['value'] = '';
+  }
   if (!empty($field['settings']['todate'])) {
-    $date = date_default_value_part($item, $field, $instance, $langcode, 'value2');
-    $item[0]['value2'] = is_object($date) ? date_format($date, $db_format) : '';
+    $date2 = date_default_value_part($item, $field, $instance, $langcode, 'value2');
+    if (is_object($date2) && empty($date2->errors)) {
+      $item[0]['value2'] = date_format($date2, $db_format);
+    }
+    else {
+      $item[0]['value2'] = '';
+    }
```

Comment:
I don't like these weirdly-named variables as we inherit them from D7. Can we please change all that to this instead?:

```suggestion
    $to_date = date_default_value_part($item, $field, $instance, $langcode, 'to_date_value');
    if (is_object($to_date) && empty($to_date->errors)) {
      $item[0]['to_date_value'] = date_format($to_date, $db_format);
    }
    else {
      $item[0]['to_date_value'] = '';
    }
```

Reply:
...never mind. Out of scope for this PR, so I've filed https://github.com/backdrop/backdrop-issues/issues/5069